doc/topics/writing-recipes.rst

```diff
@@ -337,6 +342,267 @@ entry points, using the type ``zc.buildout.uninstall`` as can be seen
+
+.. code-block:: ini
+
+   [buildout]
+   develop = src
+   parts = py
+
+   [py]
+   recipe = zc.recipe.egg
+   eggs = democonfigrecipe [test]
+   interpreter = py
+
+.. -> src
+
+    >>> testing.write('buildout.cfg', src)
+    >>> testing.run_buildout_in_process()
+
+Running Buildout with this gives is an interpreter script that we can
+run our tests with.  The script will make sure that ``zc.buildout``
+and our recipe can be imported.
+
+To run our tests:
+
+.. code-block:: console
+
+    bin/py src/testdemoconfigrecipe.py
+
+.. -> src
+
+    >>> print(testing.system(src)) # doctest: +ELLIPSIS
+    ..
+    ----------------------------------------------------------------------
+    Ran 2 tests ...
+    OK
+    <BLANKLINE>
+
+In this example, we've tried to keep things simple and as free from
+external requirements as possible.
+
+More realistically:
+
+- You'd probably arrange your recipe in a Python package rather than
+  as a top-level module and a top-level testing module.
+
+- You might use a test runner like nose or pytest.  There are `recipes
+  that can help set this up
+  <https://pypi.python.org/pypi?%3Aaction=search&term=test+runner+buildout+recipe&>`_.
+  We just used the test runner built into ``unittest``.
+
+``zc.buildout.testing`` reference
+----------------------------------
+The zc.buildout.testing module provides an API that can be used when
+writing recipe tests.  This API is documented below.
+
+Many of the functions documented below take a path argument as
+multiple arguments.  These are joined using ``os.path.join``.  This is
+more convenient than having to call os.path.join before calling the
+functions.
+
+``Buildout()``
+    A class you can use to create buildout and sections objects in your tests
+
+    This is a subclass of the main object used to run buildout.  Its
+    constructor takes no arguments.  You can add data to it by setting
+    section names to dictionaries::
+
+       buildout['config'] = dict(contents=self.config)
+
+    To get an options object to pass to your recipe, just ask for it back::
+
+       buildout['config']
+
+    See the :ref:`recipe example <recipe-example>` above.
+
+``cat(*path)``
+    Display the contents of a file.  The file path is provided as one or
+    more strings, to be joined with os.path.join.
+
+    On Windows, if the file doesn't exist, the function will try
+    adding a '-script.py' suffix.  This helps to work around a
+    difference in script generation on windows.
+
+``clear_here()``
+    Remove all files and directories in the current working directory.
+
+    *New in buildout 2.9*
+
+``eqs(got, *expected)``
+    Compare a collection with a collection given as multiple
+    arguments.
+
+    Both collections are converted to and compared as sets.  If the
+    sets are the same, then no output is returned, otherwise a tuple
+    of extras is returned, so, for example::
+
+      >>> eqs([1, 2, 3], 3, 1, 2)
+      >>> eqs([1, 2, 3], 1, 2, 4) == (set([3]), set([4]))
```

Comment:
We no longer support Python 2.6 so we can use the set syntax here for perhaps more clarity?  `== ({3}, {4})`.

OTOH that doesn't match the repr() of the sets returned, so maybe the current syntax is better.